### ISSUE TITLE
[nu-validator coverage] Validate img[usemap] → <map name> reference (hash-name resolution)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3945]
# [nu-validator coverage] Validate `img[usemap]` → `<map name>` reference (hash-name resolution)

## Repro

```html
<img src="test.png" alt="test" usemap="#nonexistent">
```

with no `<map name="nonexistent">` anywhere in the document.

nu-validator: error — "The hash-name reference in attribute 'usemap' referred to 'nonexistent', but there is no 'map' element with a 'name' attribute with that value."
markuplint (all rules enabled by the bench): 0 violations.

Fixture: `tests/external/validator/tests/html/elements/img/usemap-invalid-reference-novalid.html` (`nu-only`).

## Spec

HTML LS "Image maps — Authoring" (`image-maps.html#authoring`):

> "The `usemap` attribute, if specified, must be a valid hash-name reference to a `map` element."

HTML LS "Valid hash-name reference" (`common-microsyntaxes.html#valid-hash-name-reference`):

> "A valid hash-name reference to an element of type type is a string consisting of a U+0023 NUMBER SIGN character (#) followed by a string which exactly matches the value of the `name` attribute of an element with type type in the same tree."

Note: "exactly matches" is **string-equal** (case-sensitive). This is the authoritative definition and applies to the `<img usemap>` → `<map name>` lookup.

## Current markuplint behaviour

markuplint already has adjacent coverage for element cross-references (`no-refer-to-non-existent-id` for `id`-based references, `form-attr-references-form` for `[form]` → `<form id>`, `input-list-references-datalist` for `[list]` → `<datalist id>`, `label-for-references-labelable` for `<label for>` → labelable element).

The `<map>`-related rule `map-id-name-match` (`packages/@markuplint/rules/src/map-id-name-match`) covers the internal id/name consistency of `<map>` itself but does **not** validate `img[usemap]` → `<map name>` lookups. Fixture confirms markuplint's silence: `id` is not involved (map uses `name`), so `no-refer-to-non-existent-id` does not fire; the hash-name reference resolution logic is missing entirely.

## Proposal

Add a new rule (working name: `usemap-references-map`) that mirrors the shape of `form-attr-references-form` / `input-list-references-datalist`:

- Selector: `img[usemap]` (scope). `<object usemap>` support is a possible future extension but out of scope for the initial rule.
- Extract the hash from the attribute value, strip the leading `#`.
- Look up any `<map>` in the same tree whose `name` attribute string-equals (case-sensitive per the definition above).
- If none, report.

### Related follow-up (pre-existing spec discrepancy)

`packages/@markuplint/rules/src/map-id-name-match/index.spec.ts` `[map-id-name-match-invalid-002]` contains this comment:

> "HTML LS `<map>` requires the id and name attributes to be string-equal; the comparison is case-sensitive even though the name attribute itself is matched case-insensitively against `<img usemap>`."

The "matched case-insensitively against `<img usemap>`" claim contradicts the "exactly matches" wording verified above. This should be reconciled — either the spec has changed or the comment was wrong at authoring time — separately from this Issue's proposed rule. Flag but do not require alignment as a blocker for this Issue.

Overlap check: `id`-name coexistence is common (`<map id=foo name=foo>`) so a naive `usemap="#foo"` might also flag `no-refer-to-non-existent-id`. The two rules should coexist without duplication — `no-refer-to-non-existent-id` targets `id`-typed attributes only, so `usemap` (declared as `HashName` in `spec.img.jsonc`) is out of its scope.

## Priority

Deferred.

## Implementation notes

- A default-on `usemap-references-map` rule will flag existing pages with typos or removed maps — the implementation PR must include release-note wording.
- This Issue is registered as the `primary` mapping for the fixture in `tests/external/bench/issue-xref.config.ts`; xref refresh (`yarn bench:xref`) is the source of truth for the current fixture set backing this Issue.

## Benchmark cross-reference

_Sync'd from `tests/external/snapshots/` (submodule=1429313, nu-validator@sha256:210bbc353b23, markuplint=5.0.0-rc.4, generated=2026-08-12 03:50:09 UTC)._

Filter: `^html\/elements\/img\/usemap-invalid-reference-novalid` → **1** fixtures. Verdict tally: match-error=1, match-clean=0, ml-only=0, nu-only=0, nu-over=0.

[HTML LS "Image maps — Authoring"](https://html.spec.whatwg.org/multipage/image-maps.html#authoring): "The `usemap` attribute, if specified, must be a valid hash-name reference to a `map` element." [HTML LS "Valid hash-name reference"](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#valid-hash-name-reference): the reference is a `#`-prefixed string that "exactly matches" a `name` attribute in the same tree (string-equal, case-sensitive). `packages/@markuplint/rules/src/map-id-name-match` covers the internal id/name consistency of `<map>` itself but does not validate `img[usemap]` → `<map name>` lookups. `no-refer-to-non-existent-id` targets `id`-typed attributes only, so `usemap` (declared as `HashName` in `spec.img.jsonc`) is out of its scope. Needs a new rule (working name `usemap-references-map`) shaped like `form-attr-references-form` / `input-list-references-datalist`. Follow-up: reconcile the `[map-id-name-match-invalid-002]` spec comment which claims `<map name>` is matched case-insensitively against `<img usemap>` — this contradicts the "exactly matches" wording verified above.

| fixture | verdict | nu id / ml rule |
| --- | --- | --- |
| `html/elements/img/usemap-invalid-reference-novalid.html` | match-error | — |

Reproduce: see [`bench-triage` skill](https://github.com/markuplint/markuplint/blob/dev/.claude/skills/bench-triage/SKILL.md) for the audit workflow.
